src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Source.cs

```diff
@@ -42,7 +43,22 @@ private static void FreeSymbolMap(MultiDictionary<string, ISymbol> symbolMap)
             return result;
         }
 
-        public static async Task<Checksum> GetSourceSymbolsChecksumAsync(Project project, CancellationToken cancellationToken)
+        /// <summary>
+        /// Cache of project to the checksum for it so that we don't have to expensively recompute
+        /// this each time we get a project.
+        /// </summary>
+        private static ConditionalWeakTable<Project, AsyncLazy<Checksum>> s_projectToSourceChecksum =
```

Comment:
I don't want it to change if dependent projects change.  i only want it to cahnge if something changes that could affect the source-symbol-tree-index.